mm2src/coins/lightning/ln_filesystem_persister.rs

```diff
@@ -0,0 +1,396 @@
+        path
+    }
+
+    pub fn manager_path(&self) -> PathBuf {
+        let mut path = self.main_path();
+        path.push("manager");
+        path
+    }
+}
+
+#[derive(Clone)]
+pub struct LightningPersisterShared(pub Arc<LightningFilesystemPersister>);
+
+impl Deref for LightningPersisterShared {
+    type Target = LightningFilesystemPersister;
+    fn deref(&self) -> &LightningFilesystemPersister { self.0.deref() }
+}
+
+impl Persister<InMemorySigner, Arc<ChainMonitor>, Arc<Platform>, Arc<KeysManager>, Arc<Platform>, Arc<LogState>>
+    for LightningPersisterShared
+{
+    fn persist_manager(&self, channel_manager: &ChannelManager) -> Result<(), std::io::Error> {
+        FilesystemPersister::persist_manager(self.0.main_path().display().to_string(), channel_manager)?;
+        if let Some(backup_path) = self.0.backup_path() {
+            let file = fs::OpenOptions::new()
+                .create(true)
+                .write(true)
+                .truncate(true)
+                .open(backup_path)?;
+            channel_manager.write(&mut BufWriter::new(file))?;
+        }
+        Ok(())
+    }
+
+    fn persist_graph(&self, network_graph: &NetworkGraph) -> Result<(), std::io::Error> {
+        if FilesystemPersister::persist_network_graph(self.0.main_path().display().to_string(), network_graph).is_err()
+        {
+            // Persistence errors here are non-fatal as we can just fetch the routing graph
+            // again later, but they may indicate a disk error which could be fatal elsewhere.
+            eprintln!("Warning: Failed to persist network graph, check your disk and permissions");
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(target_os = "windows")]
+macro_rules! call {
+    ($e: expr) => {
+        if $e != 0 {
+            return Ok(());
+        } else {
+            return Err(std::io::Error::last_os_error());
+        }
+    };
+}
+
+#[cfg(target_os = "windows")]
+fn path_to_windows_str<T: AsRef<OsStr>>(path: T) -> Vec<winapi::shared::ntdef::WCHAR> {
+    path.as_ref().encode_wide().chain(Some(0)).collect()
+}
+
+fn write_monitor_to_file<ChannelSigner: Sign>(
+    mut path: PathBuf,
+    filename: String,
+    monitor: &ChannelMonitor<ChannelSigner>,
+) -> std::io::Result<()> {
+    // Do a crazy dance with lots of fsync()s to be overly cautious here...
+    // We never want to end up in a state where we've lost the old data, or end up using the
+    // old data on power loss after we've returned.
+    // The way to atomically write a file on Unix platforms is:
+    // open(tmpname), write(tmpfile), fsync(tmpfile), close(tmpfile), rename(), fsync(dir)
+    path.push(filename);
+    let filename_with_path = path.display().to_string();
+    let tmp_filename = format!("{}.tmp", filename_with_path);
+
+    {
+        let mut f = fs::File::create(&tmp_filename)?;
+        monitor.write(&mut f)?;
+        f.sync_all()?;
+    }
+    // Fsync the parent directory on Unix.
+    #[cfg(not(target_os = "windows"))]
```

Comment:
You can check `target_family` if it is "unix". I think it makes more sense here.

Reply:
Done